src/plugins/intel_cpu/src/graph_optimizer.cpp

```diff
@@ -507,6 +512,103 @@ void GraphOptimizer::FuseFCAndWeightsDecompression(Graph &graph) {
     }
 }
 
+void GraphOptimizer::FuseGatherAndWeightsDecompression(Graph &graph) {
+    std::set<ov::element::Type> supportedWeightsPrecisions{ov::element::u8};
```

Comment:
Even though U8 is default precision for embeddings in NNCF, it still has an ability to compress them into U4 precision, which is crucial for best memory footprint. 
@xipingyan  As a follow-up activity we need to extend support on U4 precision as well.

Reply:
@yuxu42 FYI

Reply:
New PR will process precision U8 case.